Automattic/wp-calypso
# Settings: Refactor Post Formats to use `react-query`

#### Changes proposed in this Pull Request

Considering that post formats a very straightforward example of server state on the client side in Calypso, and the simplicity of the use case (read-only), this makes it a great example of how much boilerplate we can remove by using `react-query` instead of redux.

This PR refactors the post formats Redux state to use `react-query` and ports the one single usage of this state tree.

#### Testing instructions

* Make sure your WP.com site is using a theme that supports one or more custom post formats. You can find such a theme here: https://wordpress.com/themes/filter/post-formats/:site where `:site` is your WP.com site.
* Go to `/settings/writing/:site`
* Verify the post formats in the "Default post format" are loading correctly.
* Pick a site with a theme without any custom post formats. 
* Verify the field correctly displays just the "Standard" option and there are no errors in the console.
